interface/java_binding/src/main/cpp/libktx-jni.cpp

```diff
@@ -97,12 +97,14 @@ void copy_ktx_astc_params(JNIEnv *env, jobject params, ktxAstcParams &out)
     out.normalMap = env->GetBooleanField(params, normalMap);
     out.perceptual = env->GetBooleanField(params, perceptual);
 
-    env->GetByteArrayRegion(
-        static_cast<jbyteArray>(env->GetObjectField(params, inputSwizzle)),
-        0,
-        4,
-        reinterpret_cast<jbyte*>(&out.inputSwizzle)
-    );
+    jobject inputSwizzleObject = env->GetObjectField(params, inputSwizzle);
```

Comment:
Having the fix from #876 here is going to cause conflicts come merge time. Does this PR need the swizzle fix? If not, please remove it.

Reply:
I'll go through some details later today, but only a quick note here: 

The statement
> Builds upon https://github.com/KhronosGroup/KTX-Software/pull/876 
implies: It is really _based on the state_ (commit hash) from the swizzle PR. So when the swizzle PR is merged, there will be no conflict here. And even when the swizzle PR is closed (unmerged), and only _this_ is merged, then the result would still include the swizzle fix. 

I opened this PR - which goes _beyond_ the swizzle fix - as a 'Draft', because there may be a few more details to talk about before merging it. 

(And eventually, even _this_ PR might become obsolete, depending on how the https://github.com/KhronosGroup/KTX-Software/issues/880 will be tackled, but I'll try to create a rough "work plan" based on your other comments)

Reply:
Thanks for the explanation. Makes sense.